src/brent.ts

```diff
@@ -0,0 +1,341 @@
+             *  for this point
+             */
+            data: best.data,
+            /** The number of times that `target` has been called so far */
+            evaluations: this._evaluations,
+            /** The number of times that {@link Brent.step} has been
+             *  called so far
+             */
+            iterations: this._evaluations - 1,
+            /** The best found location */
+            location: best.location,
+            /** The value of `target(location)` */
+            value: this._control.findMax ? -best.value : best.value,
+        };
+    }
+
+    private _point(location: number): Point1 {
+        this._evaluations++;
+        const result = checkResult(this._target(location));
+        return {
+            data: result.data,
+            location,
+            value: result.value,
+        };
+    }
+
+    private _step() {
+        const state = this._state;
+        const xm = (state.a + state.b) / 2;
+        const x = state.x.location;
+        const tol1 = sqrtMachineEps * Math.abs(x) +
+            this._control.tolerance / 3.0;
+        const tol2 = 2.0 * tol1;
+
+        // Test to see if we've converged
+        if (Math.abs(x - xm) <= (tol2 - 0.5 * (state.b - state.a))) {
+            return true;
+        }
+
+        const [p, q] = fitParabola(state, tol1);
+
+        const useGoldenRatio =
+            Math.abs(p) >= Math.abs(q * 0.5 * state.e) ||
+            p <= q * (state.a - x) ||
+            p >= q * (state.b - x);
+
+        if (useGoldenRatio) {
+            if (x < xm) {
+                state.e = state.b - x;
+            } else {
+                state.e = state.a - x;
+            }
+            state.d = squaredInverseGoldenRatio * state.e;
+        } else { // Parabolic interpolation
+            state.e = state.d;
+            state.d = p / q;
+            const uNext = x + state.d; // candidate u for below
+            // f must not be evaluated too close to a or b (lower or upper)
+            if (uNext - state.a < tol2 || state.b - uNext < tol2) {
+                state.d = x < xm ? tol1 : -tol1;
+            }
+        }
+
+        // f must not be evaluated too close to x
+        const d = Math.abs(state.d) >= tol1 ? state.d : copysign(tol1, state.d);
+        const u = this._point(x + d);
+
+        updateState(u, state);
+        return false;
+    }
+}
+
+interface Point1 {
+    readonly data: any;
+    readonly location: number;
+    readonly value: number;
+}
+
+/**
+ * See p 73 of the book
+ *
+ * We track a set of points on the domain [a, b, u, v, w, x], not all
+ * distinct.
+ *
+ * * Initially (`a`, `b`) is the domain of `target` where we will search
+ *   for the minimum.
+ * * `x` is the point visited with the lowest value of `target`
+ * * `w` is the second lowest visited value of `target`
+ * * `v` is the previous value of `w`
+ * * `fx`, `fw` and `fv` are the values of `target` evaluated at `x`, `w`
```

Comment:
Is this just explanatory, or was the intention to include fx etc in state?

Reply:
this was before a refactor - there used to be elements x, w, v, fx, fw, fv but I merged them into the `Point1` interface which simplified the bookkeeping. Will update the comment